### PR TITLE
Added a blog post about authenticating ejb-over-http

### PR DESCRIPTION
**Pull request by Ashpan**

### CODE REVIEW

**Comment by Ashpan** [comment]
https://github.com/wildfly-security-incubator/elytron-examples/pull/44

**Comment by Ashpan** [comment]
https://ashpan.github.io/wildfly-elytron/blog/ejb-over-http/

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+[source]
+----
+/subsystem=ejb3/application-security-domain=other:add(security-domain=fsSD)
+----
+
+== Configure the HTTP Transport
+
+First we must configure the http-authentication-factory for the server to authenticate over HTTP.
+Here we set the mechanism to BASIC (but could also be set to FORM), point to the fsSD security-domain, and the fsRealm filesystem-realm, and we call the newly created http-authentication-factory, example-http-auth
```
Comment (fjuma) [inline _posts/2021-06-09-ejb-over-http.adoc:62]: "(but could also be set to FORM)" ->

"(but could also be set to another HTTP mechanism, like FORM for example)"

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+----
+
+Then in order to operate over HTTP, we must disable the legacy security-realm. This is done by undefining it from the http-invoker
+
+[source]
+----
+/subsystem=undertow/server=default-server/host=default-host/setting=http-invoker:undefine-attribute(name=security-realm)
+----
+
+And finally we can enable the http-invoker with the http-authentication-factory defined earlier (example-http-auth) that allows remote HTTP invocation for the EJB
```
Comment (fjuma) [inline _posts/2021-06-09-ejb-over-http.adoc:76]: "remote HTTP" -> "HTTP"

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+== Configure the Credential Store
+
+To create a credential store, we run the following command, which creates it in the path specified after the --location flag, and secures it with the password specified after the --password flag (in this case it is “StorePassword”)
+
+[source]
+----
+WILDFLY_HOME/bin/elytron-tool.sh credential-store --create --location "/PATH/TO/mycredstore.cs" --password StorePassword
+----
+
+We then create an alias to be able to access this credential store with a different username and password, in this case the username being “quickstartUser”, and the password being “quickstartPwd1!”
```
Comment (fjuma) [inline _posts/2021-06-09-ejb-over-http.adoc:92]: "We then create an alias to be able to access this credential store with a different username and password"

I think this should be reworded to indicate that we're adding an alias (quickstartUser) to the credential store that will be used to hold the password that we want to use when accessing the EJB.

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            <clear-password password="StorePassword"/>
+        </protection-parameter-credentials>
+    </credential-store>
+</credential-stores>
+----
+Here we can see the definition of the credential store.
+Make sure to change the path from "/PATH/TO/mycredstore.cs", to the same path as when configuring the credential store. Also ensure that the path here is the absolute path, starting from `/`
+
+== Summary
+This blog post has given an overview on how to create an EJB over HTTP, authenticate it with a FileSystem realm, and use a credential-store to invoke the EJB.
```
Comment (fjuma) [inline _posts/2021-06-09-ejb-over-http.adoc:118]: "how to create" -> "how to invoke"

"authenticate it with a FileSystem realm" -> "make use of a filesystem realm"

"use a credential-store to invoke the EJB" -> "use a credential-store to obtain the password that should be used to invoke the EJB"

**Review by fjuma — COMMENTED** [review]
Thanks for the post @Ashpan! I've just added some small comments.

**Comment by fjuma** [comment]
Just noticed that when describing the client configuration, it would be good to also highlight the code from your client that configures the PROVIDER_URL to show what that needs to be set to when using EJB over HTTP.

File: _posts/2021-06-09-ejb-over-http.adoc (hunk 3 above)
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Changed to
> We now create an alias that will allow us to have access to the credential store password when accessing the EJB. In this case the alias username is “quickstartUser”, and the password is “quickstartPwd1!”:

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 5]
```diff
@@ -0,0 +1,118 @@
+---
+layout: post
+title: "Authenticating EJBs over HTTP"
+date: 2021-06-09
+tags: wildfly elytron
+synopsis: An overview of the how to setup EJBs to operate over HTTP.
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Fixed

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+The *https://github.com/wildfly-security-incubator/elytron-examples/tree/master/ejb-http[ejb-http]* quickstart will describe the steps to setting up EJBs using HTTP as the transport. It will also secure it with a FileSystem realm, and use a credential store to access the EJB. This will focus on the configuration of the server to allow it to communicate over HTTP.
+
+== Configure the filesystem realm
+First we will connect to the server via 
+
+[source, shell]
+----
+$ WILDFLY_HOME/bin/jboss-cli.sh --connect
+----
+`WILDFLY_HOME` refers to the location where Wildfly is installed. For example it could be something like `~/wildfly/build/target/wildfly-24.0.0.Beta1-SNAPSHOT`
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Fixed

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+Then we will create a file system realm called fsRealm, and add a directory in the jboss server config directory called fs-realm-users that will contain all the data of the identity of each user created
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Fixed

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+We will also add a role decoder to be able to process the roles assigned
+
+[source]
+----
+/subsystem=elytron/simple-role-decoder=from-roles-attribute:add(attribute=Roles)
+----
+
+== Configure the Security Domain
+
+We will create a security domain that resides in the filesystem realm and will use the role decoder we created above
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Fixed

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+[source]
+----
+/subsystem=elytron/security-domain=fsSD:add(realms=[{realm=fsRealm, role-decoder=from-roles-attribute}], default-realm=fsRealm,permission-mapper=default-permission-mapper)
+----
+
+We will also add the filesystem security domain mapping in the EJB3 subsystem so that the server uses the correct security domain
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Fixed

File: _posts/2021-06-09-ejb-over-http.adoc (hunk 1 above)
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Fixed

File: _posts/2021-06-09-ejb-over-http.adoc (hunk 2 above)
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Fixed

File: _posts/2021-06-09-ejb-over-http.adoc (hunk 4 above)
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Fixed

**Comment by Ashpan** [comment]
> Just noticed that when describing the client configuration, it would be good to also highlight the code from your client that configures the PROVIDER_URL to show what that needs to be set to when using EJB over HTTP.

Added

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+/subsystem=ejb3/application-security-domain=other:add(security-domain=fsSD)
+----
+
+To create a credential store, we run the following command, which creates it in the path specified after the --location flag, and secures it with the password specified after the --password flag (in this case it is “StorePassword”):
+[source]
+----
+WILDFLY_HOME/bin/elytron-tool.sh credential-store --create --location "/PATH/TO/mycredstore.cs" --password StorePassword
+----
+
+We now create an alias that will allow us to have access to the credential store password when accessing the EJB. In this case the alias username is “quickstartUser”, and the password is “quickstartPwd1!”:
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Fixed

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+    <credential-store name="mycredstore">
+        <attributes>
+            <attribute name="keyStoreType" value="JCEKS"/>
+            <attribute name="location" value="/PATH/TO/mycredstore.cs"></attribute>
+        </attributes>
+        <protection-parameter-credentials>
+            <clear-password password="StorePassword"/>
+        </protection-parameter-credentials>
+    </credential-store>
+</credential-stores>
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Added a snippet of the whole file

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+----
+Here we can see the definition of the credential store.
+Make sure to change the path from "/PATH/TO/mycredstore.cs", to the same path as when configuring the credential store. Also ensure that the path here is the absolute path, starting from `/`
+
+=== Provider URL
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Fixed

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+When creating the context in your Client Application, the `PROVIDER_URL` field should be changed.
+Originally it is: +
+`"remote+http://localhost:8080"` +
+and it should be changed to: +
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc]: Fixed

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 14]
```diff
@@ -0,0 +1,118 @@
+---
+layout: post
+title: "Invoking EJBs over HTTP"
+date: 2021-06-10
+tags: wildfly elytron
+synopsis: An overview of how to setup EJBs to operate over HTTP.
+author: araskar
+---
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc:8]: Cool, added

File: _posts/2021-06-09-ejb-over-http.adoc [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+
+The *https://github.com/wildfly-security-incubator/elytron-examples/tree/master/ejb-http[ejb-http]* quickstart will describe the steps to setting up EJBs using HTTP as the transport. It will also secure it with a FileSystem realm, and use a credential store to access the EJB. This will focus on the configuration of the server to allow it to communicate over HTTP.
+
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-06-09-ejb-over-http.adoc:15]: Added

**Review by fjuma — APPROVED** [review]
Thanks @Ashpan! Looks great.

**Pull request merged**